### ISSUE TITLE
Product image failure when importing through CSV

### ISSUE THREAD

**Issue by flytomek** [magento/magento2#20098]
### Preconditions

1. Magento 2.3.0
2. PHP Version 7.2.13
3. macOS Mojave 10.14.1
4. release with no Sample Data
5. Valid product image under remote URL `http://www.tompetty.com/sites/g/files/g2000007521/f/styles/photo-carousel/public/sample1_1.jpg`

### Steps to reproduce

1. Create CSV file
```
sku;store_view_code;attribute_set_code;product_type;categories;product_websites;name;description;short_description;weight;product_online;tax_class_name;visibility;price;special_price;special_price_from_date;special_price_to_date;url_key;meta_title;meta_keywords;meta_description;base_image;base_image_label;small_image;small_image_label;thumbnail_image;thumbnail_image_label;swatch_image;swatch_image_label;created_at;updated_at;new_from_date;new_to_date;display_product_options_in;map_price;msrp_price;map_enabled;gift_message_available;custom_design;custom_design_from;custom_design_to;custom_layout_update;page_layout;product_options_container;msrp_display_actual_price_type;country_of_manufacture;additional_attributes;qty;out_of_stock_qty;use_config_min_qty;is_qty_decimal;allow_backorders;use_config_backorders;min_cart_qty;use_config_min_sale_qty;max_cart_qty;use_config_max_sale_qty;is_in_stock;notify_on_stock_below;use_config_notify_stock_qty;manage_stock;use_config_manage_stock;use_config_qty_increments;qty_increments;use_config_enable_qty_inc;enable_qty_increments;is_decimal_divided;website_id;related_skus;related_position;crosssell_skus;crosssell_position;upsell_skus;upsell_position;additional_images;additional_image_labels;hide_from_product_page;custom_options;bundle_price_type;bundle_sku_type;bundle_price_view;bundle_weight_type;bundle_values;bundle_shipment_type;configurable_variations;configurable_variation_labels;associated_skus
b18-r1t01-w0-0m;;Default;virtual;;base;R1 anniversary print T-shirt M;;;;1;Taxable Goods;Not Visible Individually;25.9500;;;;r1-anniversary-print-t-shirt-m;R1 anniversary print T-shirt;;R1 anniversary print T-shirt;;;;;;;;;1/5/19, 1:55 PM;1/5/19, 2:04 PM;;;Block after Info Column;;;;No;;;;;;;;;size=M;999.0000;0.0000;1;0;0;1;1.0000;1;10000.0000;1;1;1.0000;1;1;1;1;1.0000;1;0;0;0;;;;;;;;;;;;;;;;;;;
b18-r1t01-w0-0l;;Default;virtual;;base;R1 anniversary print T-shirt L;;;;1;Taxable Goods;Not Visible Individually;26.9500;;;;r1-anniversary-print-t-shirt-l;R1 anniversary print T-shirt;;R1 anniversary print T-shirt;;;;;;;;;1/5/19, 1:55 PM;1/5/19, 2:04 PM;;;Block after Info Column;;;;No;;;;;;;;;size=L;999.0000;0.0000;1;0;0;1;1.0000;1;10000.0000;1;1;1.0000;1;1;1;1;1.0000;1;0;0;0;;;;;;;;;;;;;;;;;;;
b18-r1t01-w0-1l;;Default;virtual;;base;R1 anniversary print T-shirt XL;;;;1;Taxable Goods;Not Visible Individually;29.9500;;;;r1-anniversary-print-t-shirt-xl;R1 anniversary print T-shirt;;R1 anniversary print T-shirt;;;;;;;;;1/5/19, 1:55 PM;1/5/19, 2:03 PM;;;Block after Info Column;;;;No;;;;;;;;;size=XL;999.0000;0.0000;1;0;0;1;1.0000;1;10000.0000;1;1;1.0000;1;1;1;1;1.0000;1;0;0;0;;;;;;;;;;;;;;;;;;;
@b18-r1t01-w0-0l;;Default;configurable;;base;R1 anniversary print T-shirt;;;;1;Taxable Goods;Catalog, Search;;;;;r1-anniversary-print-t-shirt;R1 anniversary print T-shirt;;R1 anniversary print T-shirt;http://www.tompetty.com/sites/g/files/g2000007521/f/styles/photo-carousel/public/sample1_1.jpg;;http://www.tompetty.com/sites/g/files/g2000007521/f/styles/photo-carousel/public/sample1_1.jpg;;http://www.tompetty.com/sites/g/files/g2000007521/f/styles/photo-carousel/public/sample1_1.jpg;;;;1/5/19, 1:55 PM;1/5/19, 1:55 PM;;;Block after Info Column;;;;Use config;;;;;;;;;;0.0000;0.0000;1;0;0;1;1.0000;1;10000.0000;1;1;1.0000;1;1;1;1;1.0000;1;0;0;0;;;;;;;http://www.tompetty.com/sites/g/files/g2000007521/f/styles/photo-carousel/public/sample1_1.jpg;;;;;;;;;;sku=b18-r1t01-w0-0m,size=M|sku=b18-r1t01-w0-0l,size=L|sku=b18-r1t01-w0-1l,size=XL;size=Size;
```
2. Go to System > Import with following settings:
```
Entity Type - products
Import Behavior - add/update
Stop on error
Allowed Errors Count - 10
Field separator - ;
Multiple value separator - ,
Empty attribute value constant - __EMPTY__VALUE__
Fields enclosure - unchecked
Images File Directory - blank
```
and select CSV file

### Expected result

1. Import 3 virtual products and one configurable product with image

### Actual result

1. When importing: Following Error(s) has been occurred during importing process: 1. Imported resource (image) could not be downloaded from external resource due to timeout or access permissions in row(s): 4
2. Products are imported but configurable product has no image in Magento (however **image has been successfully uploaded** to `pub/media/import`)

![](https://i.imgur.com/TtivbWf.png)
![](https://i.imgur.com/R53nt9y.png)

**Comment by magento-engcom-team** [comment]
Hi @flytomek. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@flytomek do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by flytomek** [comment]
@magento-engcom-team give me 2.3.0 instance

**Comment by magento-engcom-team** [comment]
Hi @flytomek. Thank you for your request. I'm working on Magento 2.3.0 instance for you

**Comment by digikeijs** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @digikeijs. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@irajneeshgupta Thank you for verifying the issue.

Unfortunately, not enough information was provided to created internal ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@irajneeshgupta Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97478` were created

**Comment by flytomek** [comment]
Just quick update from me. Since I haven't received fresh instance, I set up Docker container (alexcheng/magento2 image), reproduced all steps and actual result is the same as above under my local machine.

**Comment by azue** [comment]
Magento 2.2.7 is also affected. 
@magento-engcom-team can you add a Note to the release notes https://devdocs.magento.com/guides/v2.2/release-notes/ReleaseNotes2.2.7CE.html

**Comment by magento-engcom-team** [comment]
Hi @flytomek. Thank you for your report. 
The issue has been fixed in magento/magento2#20127 by @irajneeshgupta in 2.3-develop branch
Related commit(s):
 - [7f4dffdbd5b58a39ae043a9b47a5c75e19f6a552](https://github.com/magento/magento2/commit/7f4dffdbd5b58a39ae043a9b47a5c75e19f6a552)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @flytomek. Thank you for your report. 
The issue has been fixed in magento/magento2#20329 by @irajneeshgupta in 2.2-develop branch
Related commit(s):
 - [40f64f9f70be41d0191246cf9d647d20ee4812a4](https://github.com/magento/magento2/commit/40f64f9f70be41d0191246cf9d647d20ee4812a4)

The fix will be available with the upcoming 2.2.8 release.

**Comment by flytomek** [comment]
I've set up fresh Magento 2.3.3 instance on new server and... exactly this issue happened again!
